source/common/network/connection_impl.cc

```diff
@@ -74,10 +74,9 @@ ConnectionImpl::ConnectionImpl(Event::DispatcherImpl& dispatcher, int fd,
                                Address::InstanceConstSharedPtr local_address,
                                bool using_original_dst, bool connected)
     : filter_manager_(*this, *this), remote_address_(remote_address), local_address_(local_address),
-      read_buffer_(dispatcher.getBufferFactory().create()),
-      write_buffer_(Buffer::InstancePtr{dispatcher.getBufferFactory().create()},
-                    [this]() -> void { this->onLowWatermark(); },
-                    [this]() -> void { this->onHighWatermark(); }),
+      read_buffer_(new Buffer::OwnedImpl), write_buffer_(dispatcher.getBufferFactory().create(
```

Comment:
Any reason to not make read_buffer_ an inline OwnedImpl to avoid extra allocation?